src/models/data_models/IConfiguration.ts

```diff
@@ -0,0 +1,31 @@
+export interface IConfiguration {
```

Comment:
`Configuration` seems like a very generic name. Could you change it to something more specific, such as `IDataSourceConfiguration`?